test/functions.php

```diff
@@ -0,0 +1,36 @@
+<?php
+require 'src/functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class Functions extends TestCase
+{
+    public function testDisplayDreamDetails ()
+    {
+        $array = 
+        [
+            [
+                'dream_title'=>'Finding nemo',
+                'dream_or_nightmare'=>'Nightmare',
+                'dream_description'=>'I was nemo, and no-one could find me!',
+                'dream_date'=>'2023-01-01'
+            ],
+            [
+                'dream_title'=>'Forrest Gump',
+                'dream_or_nightmare'=>'Dream',
+                'dream_description'=>'I just kept on running...',
+                'dream_date'=>'1990-01-01'
+            ]
+        ];
+        $expecteOutput = "<h3>Finding nemo</h3>
```

Comment:
I think that while the test is passing, it feels like a coincidence right now because your PHP indentation is the same in the test and the function. But if it wasn't, the test would fail because the string quotes are capturing the PHP indentation literally. I'd suggest either: having a separate string on each line and using the dot operator -
```php
$expectedOutput = '<h3>Finding nemo</h3>\n'
    . '        <h4>Nightmare</h4>\n'
    . etc...
    . '        <p>I just kept on running...</p>\n';
```
, or looking up heredoc syntax on Google.

Reply:
and changing that in displayDreamDetails too.